test/test_linearizer.py

```diff
@@ -1075,6 +1081,120 @@ def test_local_and_grouped_reduce_multireduce(self):
       [Opt(OptOps.LOCAL, 0, 2), Opt(OptOps.GROUPTOP, 0, 2), Opt(OptOps.UPCAST, 0, 8), Opt(OptOps.UNROLL, 1, 4)],
     ])
 
+  @unittest.skip("multireduce isn't supported yet")
+  @unittest.skipUnless(Device[Device.DEFAULT].renderer.has_local, "test requires locals")
+  def test_just_enough_barriers(self):
```

Comment:
a quick check to make sure there are barriers between local buffer reads/writes and there aren't duplicate buffers

Reply:
can these very specific tests be replaced with correctness tests?
Ideally we want np.testing.assert_allclose to catch everything.

Reply:
the asts/kernels they generate are tested for correctness in the other tests. this is to catch if there are duplicate barriers, which doesn't break correctness but we don't want all the same

I'm happy to move this check into the other tests though

Reply:
If correctness doesn't catch this it's ok.